File: backend/WeatherForecast.SharedTypes/UIDto/WeatherForecastDto.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Appsfactory.WeatherForecast.SharedTypes.UIDto
{
    [DebuggerDisplay("{Date}.{Temperature} ºC - {Humidity}, Wind speed: {WindSpeed}")]
    public record WeatherForecastDto
    {
        [JsonPropertyName("date")]
        public DateTime Date { get; init; }

        [JsonPropertyName("temperature")]
        public double Temperature { get; init; }
        
        [JsonPropertyName("windspeed")]
        public double WindSpeed { get; init; }
        
        [JsonPropertyName("humidity")]
        public double Humidity { get; init; }
    }
}

File: frontend/appsfactory.weatherforecast/src/components/WeatherForecastDto.ts
export default class WeatherForecastDto {
    public date: Date;
    public temperature: number;
    public humidity: number;
    public windSpeed: number;
  
    constructor() {
      this.date = new Date();
      this.temperature = 0;
      this.humidity = 0;
      this.windSpeed = 0;
    }
  }